release/models/network-instance/openconfig-network-instance.yang

```diff
@@ -1146,12 +1155,14 @@ module openconfig-network-instance {
       description
         "The type of network instance. The value of this leaf
         indicates the type of forwarding entries that should be
-        supported by this network instance. Signalling protocols
-        also use the network instance type to infer the type of
-        service they advertise; for example MPLS signalling
-        for an L2VSI network instance would infer a VPLS service
-        whereas a type of L2PTP would infer a VPWS (pseudo-wire)
-        service";
+        supported by this network instance. Signalling protocols also
+        use the network instance type to infer the type of service
+        they advertise; for example MPLS signalling for an L2VSI
+        network instance would infer a VPLS service whereas a type of
+        L2PTP would infer a VPWS (pseudo-wire) service.
+
+        An implementation must support only a single network-instance
+        of type 'DEFAULT_INSTANCE'.";
```

Comment:
This makes sense to me.  I struggle to find a way to programmatically define this constraint in yang.  I'd like to see some comment from the community if this constraint is acceptable.  In my limited knowledge, devices only support one default VRF/AFT.

Reply:
If we think we want to implement this as an actual model enforcement, we can apply a `must` constraint at the top-level `network-instances` container like:

```
    must "not(count(./network-instance/config[type='oc-netinst-types:DEFAULT_INSTANCE']) > 1)" {
      error-message "Only a single network-instance of type DEFAULT_INSTANCE is permitted";
    }
```

Reply:
I'd propose that this is where we use feature profiles/tests against the schema to implement these kind of constraints. We've generally stuck with the philosophy that the YANG here is constrained to defining the schema, and not necessarily complex inter-field validation, since we then end up mandating that there is complex XPATH parsing.

I think there's also concerns here (for me) that we make OpenConfig more brittle by doing this, since we start to try and describe more different constraints in the schema, that might not be valid for every implementation. (I do not think that this is one, but think there are cases where this will be more blurry.)

Reply:
My opinion: +1, I like the approach of using of code to implement 'branching logic constraints' vs. in the schema.

Reply:
agreed - will leave this out of schema constraints but are we in agreement of the wording used to give guidance and thus backend/client enforcement?  Loosely allowing the possibility of > 1 instances of type `DEFAULT_INSTANCE` is a recipe for disaster especially given other constraints and assumptions throughout the model sets imo

Reply:
SGTM